biomejs/biome-vscode
# feat: implement downloader

This PR implements a downloader that prompts users to download the Biome CLI if none is found. It is meant to replace the current strategy of bundling the CLI with the extension with every release.

https://github.com/biomejs/biome-vscode/assets/649677/c7c1bf81-10a5-4cd6-bbdf-019d983a2d6a

### Testing

1. `pnpm install`
2. `F5`
3. Create a new folder with a `biome.json` file inside
4. Open that folder
5. Click on the `biome.json` file
6. A notification stating that biome is not installed should appear. click download
7. When biome is installed, try formatting a file
8. Click on the biome status bar, downgrade the version, the version should change


### Fair warning

The code is admittedly a bit spaghetti, but I think it will do fine for now. We'll have to overhaul the majority of the logic if we want to support testing and more features in the future.
